File: supreme-memory/java-to-python/get_cookie.py
#!/usr/bin/env python
import requests


class GetToken(object):
    @classmethod
    def main(cls, args):
        username = "xxxxx"
        password = "xxxxx"
        auth = "{\"name\":\"" + username + "\", \"pass\": \"" + password + "\"}"
        r1 = requests.post('https://quip.bmi.stonybrook.edu/user/login?_format=json', data=auth)
        # print('TEXT', response.text) # It's just the main page.
        print('HEADERS', r1.headers)
        print('COOKIE', r1.headers['Set-Cookie'])
        r2 = requests.get('https://quip.bmi.stonybrook.edu/listofimages/48?_format=json', cookies=r1.cookies)
        print(r2.json())


if __name__ == '__main__':
    import sys

    GetToken.main(sys.argv)

File: supreme-memory/java-to-python/GetCookie.java
/**
 * It's just a tiny block of json posted to /user/login
 * At this point, your http client would have a session cookie.
 * Nothing else should have to be changed.
 */
import org.eclipse.jetty.client;
// etc...

class GetToken {
    public static void main(String[] args) {
        String server = "quip_server";
        String url = server + "/user/login?_format=json";

        String username = "";
        String password = "";
        String auth = "{\"name\":\"" + username + "\", \"pass\": \"" + password + "\"}";

        try {
            response = wc2.newRequest(url).method("POST")
                    .content(new StringContentProvider(auth), "application/json").timeout(600, TimeUnit.SECONDS)
                    .header(HttpHeader.CONTENT_TYPE, "application/json")
                    .send();
        } catch (InterruptedException | TimeoutException | ExecutionException ex) {
            Logger.getLogger(Server.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
}
